### ISSUE TITLE
Unable to parse Flutter version 

### ISSUE THREAD

**Issue by passsy** [dependabot/dependabot-core#5283]
**Package ecosystem**
dart
**Package manager version**
pub

I was testing https://github.com/dependabot/dependabot-core/pull/5207 when I ran into

```
updater | ERROR <job_394564893> dependency_services failed: Unhandled exception:
updater | <job_394564893> FormatException: Could not parse "2.18.0 (build 2.18.0-44.1.beta)".
updater | <job_394564893> #0      new Version.parse (package:pub_semver/src/version.dart:120)
updater | <job_394564893> #1      new DartSdk.<anonymous closure> (package:pub/src/sdk/dart.dart:44)
updater | <job_394564893> #2      new DartSdk (package:pub/src/sdk/dart.dart:45)
updater | <job_394564893> #3      sdk (package:pub/src/sdk.dart:56)
updater | <job_394564893> #4      PubCommand.run (package:pub/src/command.dart)
updater | <job_394564893> #5      CommandRunner.runCommand (package:args/command_runner.dart:209)
updater | <job_394564893> #6      _DependencyServicesCommandRunner.run (file:///opt/pub/helpers/bin/dependency_services.dart:60)
updater | <job_394564893> #7      main (file:///opt/pub/helpers/bin/dependency_services.dart:77)
updater | <job_394564893> #8      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295)
updater | <job_394564893> #9      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192)
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/helpers.rb:173:in `block (3 levels) in run_dependency_services'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/helpers.rb:166:in `chdir'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/helpers.rb:166:in `block (2 levels) in run_dependency_services'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/shared_helpers.rb:168:in `with_git_configured'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/helpers.rb:156:in `block in run_dependency_services'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/shared_helpers.rb:49:in `block in in_a_temporary_directory'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/shared_helpers.rb:49:in `chdir'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/shared_helpers.rb:49:in `in_a_temporary_directory'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/helpers.rb:149:in `run_dependency_services'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/helpers.rb:32:in `dependency_services_list'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/file_parser.rb:30:in `list'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-pub-0.193.0/lib/dependabot/pub/file_parser.rb:17:in `parse'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:569:in `dependencies'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:56:in `run'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/lib/dependabot/update_files_job.rb:17:in `perform_job'
updater | ERROR <job_394564893> /home/dependabot/dependabot-updater/lib/dependabot/base_job.rb:35:in `run'
updater | ERROR <job_394564893> bin/update_files.rb:22:in `<main>'
```

PR can be found here https://github.com/passsy/flutter_dependabot_example/pull/6/

Dependabot run https://github.com/passsy/flutter_dependabot_example/network/updates/394564893

Dependabot fails because dart returns `2.18.0 (build 2.18.0-44.1.beta)` as version but the parser expects to get a semver version (`2.18.0`) without the exact build number in parentheses. This seems to happen for dart preview versions.

cc @sigurdm

**Comment by jonasfj** [comment]
@passsy was this fixed by #5286 ?

**Comment by jeffwidman** [comment]
I'm going to close, assuming this was fixed by #5286 per @jonasfj helpful comment above.

If for some reason it's still an issue, please comment and we can re-open.
